### PR TITLE
[CLI-552] Offline sync large file support

### PR DESCRIPTION
**Pull request by farizrahman4u**
https://wandb.atlassian.net/browse/CLI-552

### CODE REVIEW

**Review by vanpelt — COMMENTED** [review]
This logic look correct, I wonder if we could handle this earlier as we're queueing up chunks and call _send before we hit 50mb of total chunks.  It would likely make the code simpler and would be more performant.  If that's too tricky, this will do.

File: wandb/util.py [hunk 1]
```diff
@@ -1229,3 +1235,93 @@ def _is_kaggle():
         os.getenv("KAGGLE_KERNEL_RUN_TYPE") is not None
         or "kaggle_environments" in sys.modules  # noqa: W503
     )
+
+
+def split_files(files, MAX_MB=10):
```
Comment (vanpelt) [inline wandb/util.py:1240]: Let's add a doc string explaining what `files` is and why this function exists.

**Review by vanpelt — APPROVED** [review]

File: wandb/sdk/internal/file_stream.py [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
-import os
+
+from six.moves import queue
+
+
+PY3 = sys.version_info.major == 3 and sys.version_info.minor >= 6
+if PY3:
+    from wandb.sdk.lib.file_stream_utils import split_files
+else:
+    from wandb.sdk_py27.lib.file_stream_utils import split_files
```
Comment (raubitsj) [inline wandb/sdk/internal/file_stream.py:23]: from ..lib import file_stream_utils

**Review by raubitsj — APPROVED** [review]
Great!

File: wandb/util.py [hunk 3]
```diff
@@ -755,9 +758,12 @@ def request_with_retry(func, *args, **kwargs):
             ):
                 logger.info("Rate limit exceeded, retrying in %s seconds" % delay)
             else:
+                pass
                 logger.warning(
-                    "requests_with_retry encountered retryable exception: %s. args: %s, kwargs: %s",
+                    "requests_with_retry encountered retryable exception: %s. func: %s, response: %s, args: %s, kwargs: %s",
                     e,
+                    func,
+                    e.response.content,
```
Comment (gwenzek) [inline wandb/util.py:766]: I don't think we always have `e.response`. 
In my case  request is throwing `urllib3.exceptions.ProtocolError: ('Connection aborted.', ConnectionResetError(104, 'Connection reset by peer'))`
Which is a retryable exception but `e.response` is not set , making a `e.response.content` and `AttributeError`.

Comment (vanpelt) [reply wandb/util.py:766]: Thanks for this investigation!  We'll take a look.

**Pull request merged**